skimage/measure/tests/test_regionprops.py

```diff
@@ -308,6 +306,33 @@ def test_centroid_3d():
     centroid = regionprops(SAMPLE_3D, spacing=spacing)[0].centroid
     assert_array_almost_equal(centroid, (cZ, cY, cX))
 
+    spacing = (2., 2., 2.)
+    Mpqr = get_moment3D_function(SAMPLE_3D, spacing=spacing)
+    cZ = Mpqr(1, 0, 0) / Mpqr(0, 0, 0)
+    cY = Mpqr(0, 1, 0) / Mpqr(0, 0, 0)
+    cX = Mpqr(0, 0, 1) / Mpqr(0, 0, 0)
+
+    spacing = 2
+    centroid = regionprops(SAMPLE_3D, spacing=spacing)[0].centroid
+    assert_array_almost_equal(centroid, (cZ, cY, cX))
+
+    spacing = (2.1, 2.1, 2.1)
+    Mpqr = get_moment3D_function(SAMPLE_3D, spacing=spacing)
+    cZ = Mpqr(1, 0, 0) / Mpqr(0, 0, 0)
+    cY = Mpqr(0, 1, 0) / Mpqr(0, 0, 0)
+    cX = Mpqr(0, 0, 1) / Mpqr(0, 0, 0)
+
+    spacing = 2.1
+    centroid = regionprops(SAMPLE_3D, spacing=spacing)[0].centroid
+    assert_array_almost_equal(centroid, (cZ, cY, cX))
+
+    spacing = "bad input"
+    with pytest.raises(ValueError):
+        centroid = regionprops(SAMPLE_3D, spacing=spacing)[0].centroid
+
+    spacing = ("bad input", 3.3, 4.4)
+    with pytest.raises(ValueError):
+        centroid = regionprops(SAMPLE_3D, spacing=spacing)[0].centroid
```

Comment:
Could you add these new test cases into separate test functions? That can help a lot with understanding the intention behind a test down the line and helps with keeping pytests output concise. Let me know if you need help with that. :)

Reply:
I think this does it.